### PR TITLE
Fix memory leak in getDirectoriesFoundInManifest()

### PR DESCRIPTION
**Pull request by triforce**
Please issue pull request against the `3.x-dev` branch only.

Piwik 2 is in LTS mode. This means we do not accept any pull request for 2.x except critical security bugs and major data loss bugs. 

If you need to create a pull request for 2.x, then please also create the pull request against the `3.x-dev` so we can merge both.

Happy hacking!

### CODE REVIEW

File: core/FileIntegrity.php [hunk 1]
```diff
@@ -255,13 +255,12 @@ protected static function getDirectoriesFoundInManifest()
             $directory = $file;
 
             // add this directory and each parent directory
-            while( ($directory = dirname($directory)) && $directory != '.' ) {
+            if( ($directory = dirname($directory)) && $directory != '.' ) {
```
Comment (mattab) [inline core/FileIntegrity.php]: The `while` was on purpose as it's supposed to add each parent directory to the array. Could you try to output the $directory variable and exit after like 10 or 15 iterations to see what's happening?

**Comment by mattab** [comment]
this refs #11197

**Comment by Scheirle** [comment]
With:
```
            $count = 0;
            // add this directory and each parent directory
            while( ($directory = dirname($directory)) && $directory != '.' ) {
                $count = $count + 1;
                if ($count > 20) {
                    foreach($directories as $dic) {
                        echo $dic . "<br>";
                    }
                    break;
                }
                $directories[] = $directory;
            }

```

It outputs http://pastebin.com/dteAKnjm
If I count to `40` it outputs: http://pastebin.com/hy6fpqmx

Seems like `dirname("/")` returns `/` and therefore we have a endless loop.
Adding `&& $directory != '/'` to the while loop condition works fine here.

**Comment by triforce** [comment]
Done.

**Comment by mattab** [comment]
Thank you very much for the info & pull request fix!

**Pull request merged**